ethereum/contracts/BeefyLightClient.sol

```diff
@@ -543,23 +566,102 @@ contract BeefyLightClient {
         );
     }
 
-    function createCommitmentHash(Commitment calldata commitment)
-        public
+    function encodeCommitment(Commitment calldata commitment)
+        internal
         pure
-        returns (bytes32)
+        returns (bytes memory)
     {
-        return
-            keccak256(
-                abi.encodePacked(
-                    commitment.payload,
-                    commitment.blockNumber.encode32(),
-                    commitment.validatorSetId.encode64()
-                )
-            );
+        bytes memory payload;
+        if (commitment.payload.length == 1 && commitment.payload[0].data.length == 32) {
+            payload = encodePayloadFast(commitment.payload[0]);
+        } else {
```

Comment:
hmm would an alternative solution like this work:
 - we submit Commitment in the same format as is currently, but also submit "commitmentPrefixBytes" and "commitmentSuffixBytes" and then encodeCommitment just does prefix+encodeCommitment+suffix ?

i guess it does mean that the validators could potentially sign multiple different/clashing commitments within the same signature, but it may be safe to assume that this will be handled/slashed on the polkadot side if so